### PR TITLE
simplify Command and Option constructors

### PR DESCRIPTION
**Pull request by jonsequitur**
This one contains a number of breaking changes. I've simplified the constructors for `Option` and `Command` to remove most of the optional parameters that can be set via properties. 

I welcome opinions about whether this is an improvement. Also, should the `description` parameter be removed as well? Should some of these constructor parameters be kept?

### CODE REVIEW

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs [hunk 1]
```diff
@@ -206,10 +213,23 @@ public void Argument_parses_as_the_default_value_when_the_option_has_not_been_ap
...
-            });
+            IReadOnlyCollection<Symbol> symbols = new[] {
+                new Option("-x")
+                {
+                    Argument = new Argument<int>(() => 123)
+                }
+            };
+            var command1 = new Command(
+                "something",
+                ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: Is the empty description needed here?

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs [hunk 2]
```diff
@@ -365,21 +395,31 @@ public void When_zero_or_more_arguments_of_unspecified_type_are_expected_and_non
...
-                                      {
-                                          option
-                                      });
+            IReadOnlyCollection<Symbol> symbols = new[]
+            {
+                option
+            };
+            var command1 = new Command(
+                "the-command",
+                ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: Is the empty description required here?

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs [hunk 3]
```diff
@@ -597,9 +638,11 @@ public void A_default_value_with_a_custom_constructor_can_be_specified_for_a_com
         {
             var directoryInfo = new DirectoryInfo(Directory.GetCurrentDirectory());
 
-            var command = new Command(
-                "something", "",
-                argument: new Argument<DirectoryInfo>(directoryInfo));
+            Argument argument = new Argument<DirectoryInfo>(() => directoryInfo);
+            var command = new Command("something", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: Empty description can be removed to match the change to the test below.

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs [hunk 4]
```diff
@@ -631,8 +674,11 @@ public void An_option_argument_with_a_default_argument_can_be_converted_to_the_r
         [Fact]
         public void A_command_argument_with_a_default_value_can_be_converted_to_the_requested_type()
         {
-            var command = new Command("something", "",
-                                      argument: new Argument<string>("123"));
+            Argument argument = new Argument<string>(() => "123");
+            var command = new Command("something", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs [hunk 5]
```diff
@@ -840,13 +892,13 @@ public void Values_can_be_correctly_converted_to_array_of_int_without_the_parser
         [Fact]
         public void Values_can_be_correctly_converted_to_List_of_int_without_the_parser_specifying_a_custom_converter()
         {
-            var option = new Option(
-                "-x",
-                "",
-                new Argument
+            var option = new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: Empty description can be removed. Similar comment for the rest of the tests in this file.

File: src/System.CommandLine.Tests/Help/HelpBuilderTests.cs [hunk 6]
```diff
@@ -850,8 +873,14 @@ public void Options_section_includes_option_with_empty_description()
         public void Options_section_does_not_contain_option_with_HelpDefinition_that_IsHidden()
         {
             var command = new Command("the-command");
-            command.AddOption(new Option("-x", "Is Hidden", isHidden: true));
-            command.AddOption(new Option("-n", "Not Hidden", isHidden: false));
+            command.AddOption(new Option("-x", "Is Hidden")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Help/HelpBuilderTests.cs]: Any reason to not use the object initializer to add the options?

File: src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs [hunk 7]
```diff
@@ -43,11 +43,15 @@ void Execute(string name, int age)
 
             var command = new Command("command");
             command.AddOption(
-                new Option("--name",
-                           argument: new Argument<string>()));
+                new Option("--name")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs]: Use object initializer?

File: src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs [hunk 8]
```diff
@@ -89,8 +100,14 @@ void Execute(string name, int AGE)
             }
 
             var command = new Command("command");
-            command.AddOption(new Option("--NAME", argument: new Argument { Arity = ArgumentArity.ExactlyOne }));
-            command.AddOption(new Option("--age", argument: new Argument<int>()));
+            command.AddOption(new Option("--NAME")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs]: Use object initializer?

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 9]
```diff
@@ -372,22 +384,22 @@ public void Parser_root_Options_can_be_specified_multiple_times_and_their_argume
...
+                        }.FromAmong("dog", "cat", "sheep"), Description = ""
+                    },
+                    new Option(new[] { "-v", "--vegetables" })
+                    {
+                        Argument = new Argument
                         {
                             Arity = ArgumentArity.ZeroOrMore
-                        })
-                }));
+                        }, Description = ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Any reason for the empty description setter here?

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 10]
```diff
@@ -408,20 +420,20 @@ public void Options_can_be_specified_multiple_times_and_their_arguments_are_coll
...
+                {
+                    Argument = new Argument
                     {
                         Arity = ArgumentArity.ZeroOrMore
-                    }),
-                new Option(
-                    new[] { "-v", "--vegetables" },
-                    "",
-                    new Argument
+                    }, Description = ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Any reason for the empty description setter here?

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 11]
```diff
@@ -445,20 +457,20 @@ public void When_a_Parser_root_option_is_not_respecified_but_limit_is_not_reache
...
+                {
+                    Argument = new Argument
                     {
                         Arity = ArgumentArity.ZeroOrOne
-                    }),
-                new Option(
-                    new[] { "-v", "--vegetables" },
-                    "",
-                    new Argument
+                    }, Description = ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Any reason for the empty description setter here?

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 12]
```diff
@@ -567,21 +584,23 @@ public void When_an_option_is_not_respecified_but_limit_is_reached_then_the_foll
...
-                                                   new Option(
-                                                       "--inner2", "",
-                                                       new Argument
-                                                       {
-                                                           Arity = ArgumentArity.ExactlyOne
-                                                       })
-                                               });
+            var option = new Command("outer")
+            {
+                new Option("--inner1", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 13]
```diff
@@ -567,21 +584,23 @@ public void When_an_option_is_not_respecified_but_limit_is_reached_then_the_foll
...
+                {
+                    Argument = new Argument
+                    {
+                        Arity = ArgumentArity.ExactlyOne
+                    }
+                },
+                new Option("--inner2", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 14]
```diff
@@ -604,10 +623,14 @@ public void Command_with_multiple_options_is_parsed_correctly()
...
         public void Relative_order_of_arguments_and_options_within_a_command_does_not_matter()
         {
-            var command = new Command("move", argument: new Argument<string[]>())
-                         {
-                             new Option("-X", "", new Argument<string>())
-                         };
+            var command = new Command("move")
+            {
+                new Argument<string[]>(),
+                new Option("-X", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 15]
```diff
@@ -646,10 +669,17 @@ public void Original_order_of_tokens_is_preserved_in_ParseResult_Tokens(string c
         {
             var rawSplit = commandLine.SplitCommandLine();
 
-            var command = new Command("the-command", argument: new Argument<string[]>())
+            var command = new Command("the-command")
                           {
-                              new Option("--one", "", new Argument<string>()),
-                              new Option("--many", "", new Argument<string[]>()),
+                              new Argument<string[]>(),
+                              new Option("--one", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed. And option below.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 16]
```diff
@@ -715,22 +750,22 @@ public void When_nested_commands_all_accept_arguments_then_the_nearest_captures_
...
-                              new Command(
-                                  "inner", "",
-                                  argument: new Argument<string>())
-                              {
-                                  new Command(
-                                      "non-unique",
-                                      argument: new Argument<string>())
-                              }
-                          };
+            var command = new Command("outer", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 17]
```diff
@@ -715,22 +750,22 @@ public void When_nested_commands_all_accept_arguments_then_the_nearest_captures_
...
+            {
+                new Argument<string>(),
+                new Command("non-unique")
+                {
+                    new Argument<string>()
+                },
+                new Command("inner", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 18]
```diff
@@ -756,11 +792,13 @@ public void When_child_option_will_not_accept_arg_then_parent_can()
...
         public void When_parent_option_will_not_accept_arg_then_child_can()
         {
-            var command = new Command("the-command",
-                                      argument: Argument.None)
-                          {
-                              new Option("-x", "", new Argument<string>())
-                          };
+            var command = new Command("the-command")
+            {
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 19]
```diff
@@ -895,13 +943,13 @@ public void A_root_command_can_be_omitted_from_the_parsed_args()
                           {
                               new Command("inner")
                               {
-                                  new Option(
-                                      "-x",
-                                      "",
-                                      new Argument
+                                  new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 20]
```diff
@@ -915,18 +963,18 @@ public void A_root_command_can_be_omitted_from_the_parsed_args()
...
-                                      new Argument
-                                      {
-                                          Arity = ArgumentArity.ExactlyOne
-                                      })
-                              }
-                          };
+            {
+                new Command("inner")
+                {
+                    new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 21]
```diff
@@ -942,13 +990,13 @@ public void A_renamed_RootCommand_can_be_omitted_from_the_parsed_args()
                               {
                                   new Command("inner")
                                   {
-                                      new Option(
-                                          "-x",
-                                          "",
-                                          new Argument
+                                      new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 22]
```diff
@@ -967,11 +1015,14 @@ public void Absolute_unix_style_paths_are_lexed_correctly()
...
-            var parser = new Parser(new Command("rm", "",
-                                                argument: new Argument
-                                                {
-                                                    Arity = ArgumentArity.ZeroOrMore
-                                                }));
+            Argument argument = new Argument
+            {
+                Arity = ArgumentArity.ZeroOrMore
+            };
+            var parser = new Parser(new Command("rm", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 23]
```diff
@@ -987,11 +1038,14 @@ public void Absolute_Windows_style_paths_are_lexed_correctly()
...
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 24]
```diff
@@ -1116,14 +1180,25 @@ public void The_default_behavior_of_unmatched_tokens_resulting_in_errors_can_be_
...
+                {
+                    Argument = new Argument
                     {
                         Arity = ArgumentArity.ExactlyOne
-                    })});
+                    }
+                }};
+            var command1 = new Command(
+                "the-command",
+                ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParserTests.cs [hunk 25]
```diff
@@ -1136,26 +1211,26 @@ public void Argument_names_can_collide_with_option_names()
...
-                symbols: null,
-                new Argument
-                {
-                    Arity = ArgumentArity.ZeroOrMore
-                });
+            Argument argument = new Argument
+            {
+                Arity = ArgumentArity.ZeroOrMore
+            };
+            var innerCommand = new Command("inner", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParserTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 26]
```diff
@@ -25,14 +25,14 @@ public ParsingValidationTests(ITestOutputHelper output)
         public void When_an_option_accepts_only_specific_arguments_but_a_wrong_one_is_supplied_then_an_informative_error_is_returned()
         {
             var parser = new Parser(
-                new Option(
-                    "-x",
-                    "",
-                    new Argument
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 27]
```diff
@@ -86,11 +86,10 @@ public void When_a_required_argument_is_not_supplied_then_an_error_is_returned()
...
         {
             var parser = new Parser(
-                new Command("the-command", "",
-                            new[]
-                            {
-                                new Option("-x", "")
-                            }, Argument.None));
+                new Command("the-command")
+                {
+                    new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 28]
```diff
@@ -133,11 +134,13 @@ public void An_option_can_be_invalid_when_used_in_combination_with_another_optio
         [Fact]
         public void LegalFilePathsOnly_rejects_arguments_containing_invalid_path_characters()
         {
-            var command = new Command("the-command", "",
-                                      argument: new Argument
-                                                {
-                                                    Arity = ArgumentArity.ZeroOrMore
-                                                }.LegalFilePathsOnly());
+            var command = new Command("the-command", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 29]
```diff
@@ -152,11 +155,14 @@ public void LegalFilePathsOnly_rejects_arguments_containing_invalid_path_charact
...
-            var command = new Command("the-command", "",
-                                      argument: new Argument
-                                      {
-                                          Arity = ArgumentArity.ZeroOrMore
-                                      }.LegalFilePathsOnly());
+            Argument argument = new Argument
+            {
+                Arity = ArgumentArity.ZeroOrMore
+            }.LegalFilePathsOnly();
+            var command = new Command("the-command", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 30]
```diff
@@ -389,13 +406,13 @@ public void A_command_with_subcommands_is_valid_to_invoke_if_it_has_a_handler()
         public void When_an_option_is_specified_more_than_once_but_only_allowed_once_then_an_informative_error_is_returned()
         {
             var parser = new Parser(
-                new Option(
-                    "-x",
-                    "",
-                    new Argument
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 31]
```diff
@@ -409,10 +426,10 @@ public void When_an_option_is_specified_more_than_once_but_only_allowed_once_the
         public void When_arity_is_ExactlyOne_it_validates_against_extra_arguments()
         {
             var parser = new Parser(
-                new Option(
-                    "-x",
-                    "",
-                    new Argument<int>()));
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ParsingValidationTests.cs [hunk 32]
```diff
@@ -426,9 +443,10 @@ public void When_arity_is_ExactlyOne_it_validates_against_extra_arguments()
         public void When_an_option_has_a_default_value_it_is_not_valid_to_specify_the_option_without_an_argument()
         {
             var parser = new Parser(
-                new Option(
-                    "-x", "",
-                    new Argument<int>(123)));
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ParsingValidationTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/PositionalOptionTests.cs [hunk 33]
```diff
@@ -17,14 +17,14 @@ public void When_an_option_has_a_default_value_then_a_given_positional_value_sho
...
                 {
-                    new Option(
-                        "-x",
-                        "",
-                        new Argument<int>(123)),
-                    new Option(
-                        "-y",
-                        "",
-                        new Argument<int>(456))
+                    new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/PositionalOptionTests.cs]: Empty description can be removed on options here.

File: src/System.CommandLine.Tests/PositionalOptionTests.cs [hunk 34]
```diff
@@ -152,8 +164,14 @@ public void When_a_subcommand_has_options_they_can_be_positional()
                          .EnablePositionalOptions()
                          .AddCommand(new Command("subcommand")
                                      {
-                                         new Option("-anon1", "", new Argument<string>()),
-                                         new Option("-anon2", "", new Argument<string>())
+                                         new Option("-anon1", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/PositionalOptionTests.cs]: Empty description can be removed on both options

File: src/System.CommandLine.Tests/PositionalOptionTests.cs [hunk 35]
```diff
@@ -170,7 +188,10 @@ public void When_option_argument_is_provided_without_option_name_then_argument_p
         {
             var result = new CommandLineBuilder()
                          .EnablePositionalOptions()
-                         .AddOption(new Option("-a", "", new Argument<string>()))
+                         .AddOption(new Option("-a", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/PositionalOptionTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/PositionalOptionTests.cs [hunk 36]
```diff
@@ -182,9 +203,15 @@ public void When_multiple_option_arguments_are_provided_without_option_name_then
         {
             var rootCommand = new RootCommand
                               {
-                                  new Option("-a", "", new Argument<string>()),
+                                  new Option("-a", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/PositionalOptionTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ResponseFileTests.cs [hunk 37]
```diff
@@ -257,10 +263,16 @@ public void When_response_file_parse_as_space_separated_returns_expected_values(
             var responseFile = ResponseFile(input);
 
             var rootCommand = new RootCommand
-                              {
-                                  new Option("--flag", "", new Argument<string>()),
-                                  new Option("--flag2", "", new Argument<int>())
-                              };
+            {
+                new Option("--flag", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ResponseFileTests.cs]: Empty description can be removed on both options

File: src/System.CommandLine.Tests/SymbolResultTests.cs [hunk 38]
```diff
@@ -11,10 +12,10 @@ public class SymbolResultTests
         [Fact]
         public void An_option_with_a_default_value_and_no_explicitly_provided_argument_has_an_empty_arguments_property()
         {
-            var option = new Option(
-                "-x",
-                "",
-                new Argument<string>("default"));
+            var option = new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/SymbolResultTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/SymbolResultTests.cs [hunk 39]
```diff
@@ -29,13 +30,13 @@ public void An_option_with_a_default_value_and_no_explicitly_provided_argument_h
         public void Default_values_are_reevaluated_and_not_cached_between_parses()
         {
             var option =
-                new Option(
-                    "-x",
-                    "",
-                    new Argument
+                new Option("-x", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/SymbolResultTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/SymbolResultTests.cs [hunk 40]
```diff
@@ -50,10 +51,21 @@ public void Default_values_are_reevaluated_and_not_cached_between_parses()
...
         {
-            var command = new Command("the-command", "", new[] {
+            IReadOnlyCollection<Symbol> symbols = new[] {
                 new Option(
                     new[] { "-h", "--help" }, "")
-            });
+            };
+            var command1 = new Command(
+                "the-command",
+                ""
```
Comment (Keboo) [inline src/System.CommandLine.Tests/SymbolResultTests.cs]: Empty description can be removed.

File: src/System.CommandLine.Tests/ValidationMessageLocalizationTests.cs [hunk 41]
```diff
@@ -15,11 +15,14 @@ public void Default_validation_messages_can_be_replaced_in_order_to_add_localiza
...
-            var command = new Command("the-command", "",
-                                      argument: new Argument
-                                      {
-                                          Arity = ArgumentArity.ExactlyOne
-                                      });
+            Argument argument = new Argument
+            {
+                Arity = ArgumentArity.ExactlyOne
+            };
+            var command = new Command("the-command", "")
```
Comment (Keboo) [inline src/System.CommandLine.Tests/ValidationMessageLocalizationTests.cs]: Empty description can be removed.

File: src/System.CommandLine/RootCommand.cs [hunk 42]
```diff
@@ -1,29 +1,14 @@
-﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+﻿// Copyright (c) .NET Foundation an contributors. All rights reserved.
```
Comment (Keboo) [inline src/System.CommandLine/RootCommand.cs]: Typo?

File: src/System.CommandLine.Tests/Binding/TypeConversionTests.cs (hunk 1 above)
Comment (jonsequitur) [reply src/System.CommandLine.Tests/Binding/TypeConversionTests.cs]: No, this is an artifact of this having been previously required.

**Comment by tdmiller** [comment]
I think, at the very least, the alias(es) and Argument should be left in the constructor--these are the 2 main things used at the command line ;-) unless I am missing something--just getting started with this project, love what I see!!

**Comment by jonsequitur** [comment]
@tdmiller Here's my thought process on this. I removed the argument from the `Option` constructor to bring it more in line with the way `Command` is instantiated now that a `Command` can have multiple arguments. Creating a `Command` is easier to read now using collection initializer syntax. We're considering the same for `Option`, along with the allowing multiple arguments for options, though I haven't seen examples of this kind of syntax in existing CLIs.

**Comment by tdmiller** [comment]
@jonsequitur That makes total sense--thanks for the follow up.

**Pull request merged**